### PR TITLE
Search style + department links

### PR DESCRIPTION
**Pull request by alex-shen1**
# What I did
Updated the search results page so that it's better than just a bulleted list. Results are sorted/grouped by department, with departments linked to. Probably not a particularly efficient way to implement the department link finding, but the primary bottleneck is ElasticSearch's API returning stuff anyways so it's not exactly the biggest problem IMO. 

One potential issue: as of this writing, this style takes up a whole lot more vertical space compared to the old bulleted list. Having users scroll is _probably_ not desirable, so maybe there's a way to compress it? 

## Issues Addressed
#75 kinda? I've interpreted that issue as generally "improve the search page," which this covers

# Screenshots

image

### CODE REVIEW

**Comment by j-alicia-long** [comment]
I'll work on the UI for this!

**Comment by j-alicia-long** [comment]
@alex-shen1 I did some work on styling w/ bootstrap: 

image

Only issue is that I'm unable to get the total course count for the "x results" text next to the Courses header. If you look in `search.py` you'll see some commented-out code where I attempted & failed to calculate the total course count from the query. If you could look at this that'd be great.

**Comment by j-alicia-long** [comment]
Also @alex-shen1 are you still planning to work on the search filter options?

**Comment by alex-shen1** [comment]
> Also @alex-shen1 are you still planning to work on the search filter options, as shown in Tina's mockup?

Not sure; at least in the mockup there are 2 fields and as far as I know we don't even have data for requirements yet, so only one filter is possible? Seems like a lot of work for one thing that isn't particularly helpful (especially considering the aforementioned 15 result cap; it really isn't that hard to read all 15 anyways)

**Comment by j-alicia-long** [comment]
More updates on search page styling
image

**Comment by alex-shen1** [comment]
Instructor page should really be its own branch....but it links to the search page so it's here now. Might branch off to flesh out instructors more later

**Comment by j-alicia-long** [comment]
@alex-shen1 Make sure you're running `./precommit` to check tests & pylint before you push changes so your builds won't fail on Travis :)

**Comment by alex-shen1** [comment]
whoops, should be fixed now!

**Comment by j-alicia-long** [comment]
@alex-shen1 I get this bug when I go to http://localhost:8000/instructor/3291:
image

Since you just built the instructor pages, any idea what this is caused by?

**Comment by alex-shen1** [comment]
Yup — after calculating average rating/difficulty I just rounded the value...forgot to check if the number actually exists at all. I'll have that fixed shortly!

**Comment by vn6** [comment]
Looks good! One minor UI thing the "see results for" is not showing up for me
Capture

File: tcf_website/views/search.py [hunk 1]
```diff
@@ -137,7 +140,7 @@ def format_courses(results):
         course = {
             "id": result.get("_meta").get("id"),
             "title": result.get("title").get("raw"),
-            "description": result.get("description").get("raw"),
+            # "description": result.get("description").get("raw"),
```
Comment (vn6) [inline tcf_website/views/search.py:143]: remove commented code

**Review by vn6 — APPROVED** [review]
looks good!

**Pull request merged**